python/metatensor-torch/metatensor/torch/atomistic/openmm_force.py

```diff
@@ -0,0 +1,286 @@
+            if len(requested_nls) > 1:
+                raise ValueError(
+                    "The model requested more than one neighbor list. "
+                    "Currently, only models with a single neighbor list are supported "
+                    "by the OpenMM interface."
+                )
+            elif len(requested_nls) == 1:
+                self.requested_neighbor_list = requested_nls[0]
+            else:
+                # no neighbor list requested
+                self.requested_neighbor_list = None
+
+            self.check_consistency = check_consistency
+            self.dtype = STR_TO_DTYPE[self.model.capabilities().dtype]
+            self.already_warned = False
+
+        def forward(
+            self, positions: torch.Tensor, cell: Optional[torch.Tensor] = None
+        ) -> torch.Tensor:
+            # move labels if necessary
+            selected_atoms = self.evaluation_options.selected_atoms
+            if selected_atoms is not None:
+                if selected_atoms.device != positions.device:
+                    self.evaluation_options.selected_atoms = selected_atoms.to(
+                        positions.device
+                    )
+
+            if cell is None:
+                cell = torch.zeros(
+                    (3, 3), dtype=positions.dtype, device=positions.device
+                )
+
+            # create System
+            system = System(
+                types=self.atomic_types,
+                positions=positions,
+                cell=cell,
+            )
+            system = _attach_neighbors(system, self.requested_neighbor_list)
+
+            original_dtype = system.dtype
+            if self.dtype != system.dtype:
+                if not self.already_warned:
+                    model_dtype_string = dtype_to_str(self.dtype)
+                    system_dtype_string = dtype_to_str(system.dtype)
+                    warnings.warn(
+                        f"Model dtype {model_dtype_string} does not match the dtype "
+                        f"of the system {system_dtype_string}. The system will be "
+                        f"converted to {model_dtype_string} temporarily to allow the "
+                        "model to run.",
+                        stacklevel=2,
+                    )
+                    self.already_warned = True
+                system = system.to(dtype=self.dtype)
+
+            outputs = self.model(
+                [system],
+                self.evaluation_options,
+                check_consistency=self.check_consistency,
+            )
+            energy = outputs["energy"].block().values.reshape(())
+            return energy.to(original_dtype)
+
+    metatensor_force = MetatensorForce(
+        model,
+        atomic_types,
+        selected_atoms,
+        check_consistency,
+    )
+
+    # torchscript everything
+    module = torch.jit.script(metatensor_force)
+
+    # create the OpenMM force
+    force = openmmtorch.TorchForce(module)
```

Comment:
so `model.to(device)` is called by TorchForce? Could you leave a comment if this is the case, so future readers don't wonder what's happening?